core/src/main/java/org/apache/accumulo/core/client/rfile/RFileScanner.java

```diff
@@ -74,8 +79,65 @@
 
 class RFileScanner extends ScannerOptions implements Scanner {
 
+  private static class RFileScannerIteratorEnvironmentBuilder
+      extends ClientIteratorEnvironment.Builder {
+
+    public ClientIteratorEnvironment.Builder withEnvironment(ClientServiceEnvironmentImpl env) {
+      this.env = Optional.of(env);
+      return this;
+    }
+
+  }
+
+  private static class RFileScannerEnvironmentImpl extends ClientServiceEnvironmentImpl {
+
+    private final Opts opts;
+
+    public RFileScannerEnvironmentImpl(Opts opts) {
+      super(null);
+      this.opts = opts;
+    }
+
+    @Override
+    public String getTableName(TableId tableId) throws TableNotFoundException {
+      Preconditions.checkArgument(tableId == TABLE_ID, "Expected " + TABLE_ID + " obtained"
+          + " from IteratorEnvironment.getTableId(), but got: " + tableId);
+      return TABLE_NAME;
+    }
+
+    @Override
+    public <T> T instantiate(String className, Class<T> base)
+        throws ReflectiveOperationException, IOException {
+      return RFileScanner.class.getClassLoader().loadClass(className).asSubclass(base)
+          .getDeclaredConstructor().newInstance();
+    }
+
+    @Override
+    public <T> T instantiate(TableId tableId, String className, Class<T> base)
+        throws ReflectiveOperationException, IOException {
+      return instantiate(className, base);
+    }
```

Comment:
These seem similar enough to the methods in the super class that they could be dropped here.

```suggestion
```

Reply:
Removed these in 2d39e46.